From 080af0b708cfb38f33fa60059d8d6d83aa683376 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 16 Mar 2015 21:54:26 -0400
Subject: Skip control tty code for non-ttys
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
Acked-by: Serge E. Hallyn <serge.hallyn@ubuntu.com>
---
 src/lxc/attach.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index faa90fc..69dafd4 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -985,7 +985,7 @@ static int attach_child_main(void* data)
 		new_gid = options->gid;
 
 	/* setup the control tty */
-	if (options->stdin_fd) {
+	if (options->stdin_fd && isatty(options->stdin_fd)) {
 		if (setsid() < 0) {
 			SYSERROR("unable to setsid");
 			shutdown(ipc_socket, SHUT_RDWR);
