From afbffac8805c9b9d0f6df4a58418e783ffcc049e Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Fri, 3 Apr 2015 23:40:49 +0000
Subject: lxc-test-apparmor: flush the pipe before exiting child
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

to make sure the parent's read returns.

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
Acked-by: Stéphane Graber <stgraber@ubuntu.com>
---
 src/tests/aa.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tests/aa.c b/src/tests/aa.c
index b8689ac..1ab1997 100644
--- a/src/tests/aa.c
+++ b/src/tests/aa.c
@@ -47,9 +47,11 @@ static int test_attach_write_file(void* payload)
 	if (f) {
 		printf("yes\n");
 		fclose(f);
+		fflush(NULL);
 		return 1;
 	}
 	printf("no\n");
+	fflush(NULL);
 	return 0;
 }
 
