From 4515adc496b838980ddf25ad7fd429f10ed738bc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Fri, 13 Jun 2014 17:08:44 -0400
Subject: tests: Don't fail when HOME isn't defined
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
Acked-by: Serge E. Hallyn <serge.hallyn@ubuntu.com>
---
 src/tests/lxc-test-unpriv | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tests/lxc-test-unpriv b/src/tests/lxc-test-unpriv
index 139b080..5e06e23 100755
--- a/src/tests/lxc-test-unpriv
+++ b/src/tests/lxc-test-unpriv
@@ -31,7 +31,7 @@ which newuidmap >/dev/null 2>&1 || { echo "'newuidmap' command is missing" >&2;
 
 DONE=0
 cleanup() {
-	cd
+	cd /
 
 	run_cmd lxc-stop -n c1 -k
 	pkill -u $(id -u $TUSER) -9
