From 8613b770efaff330cb13daa0508fe7bc53d82db5 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@mailbox.org>
Date: Tue, 2 Feb 2016 12:06:29 +0100
Subject: lxc-ls: exit 0 when path is not found

Signed-off-by: Christian Brauner <christian.brauner@mailbox.org>
---
 src/lxc/lxc_ls.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/lxc/lxc_ls.c b/src/lxc/lxc_ls.c
index 05162df..69e8f8a 100644
--- a/src/lxc/lxc_ls.c
+++ b/src/lxc/lxc_ls.c
@@ -353,8 +353,10 @@ static int ls_get(struct ls **m, size_t *size, const struct lxc_arguments *args,
 	if (!path)
 		goto out;
 
-	if (!dir_exists(path))
+	if (!dir_exists(path)) {
+		ret = 0;
 		goto out;
+	}
 
 	/* Do not do more work than is necessary right from the start. */
 	if (args->ls_active || (args->ls_active && args->ls_frozen))
