From 8412ce14aae0e5387b6cbf1580bcc56a299135fc Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 27 Jun 2019 14:26:14 +0200
Subject: cgroups: move variables into tighter scope

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgfsng.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index b11a71152..9cffae9e9 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -2409,15 +2409,13 @@ __cgfsng_ops static bool cgfsng_setup_limits(struct cgroup_ops *ops,
 static bool cgroup_use_wants_controllers(const struct cgroup_ops *ops,
 				       char **controllers)
 {
-	char **cur_ctrl, **cur_use;
-
 	if (!ops->cgroup_use)
 		return true;
 
-	for (cur_ctrl = controllers; cur_ctrl && *cur_ctrl; cur_ctrl++) {
+	for (char **cur_ctrl = controllers; cur_ctrl && *cur_ctrl; cur_ctrl++) {
 		bool found = false;
 
-		for (cur_use = ops->cgroup_use; cur_use && *cur_use; cur_use++) {
+		for (char **cur_use = ops->cgroup_use; cur_use && *cur_use; cur_use++) {
 			if (strcmp(*cur_use, *cur_ctrl) != 0)
 				continue;
 
