From 5770749f97d92040421fef36713a6e349c9d0ea8 Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Thu, 9 Oct 2014 11:24:37 -0500
Subject: lxc-test-unpriv: test for different cgroups per subsystem

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 src/tests/lxc-test-unpriv | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/src/tests/lxc-test-unpriv b/src/tests/lxc-test-unpriv
index a27857c..93c91a9 100755
--- a/src/tests/lxc-test-unpriv
+++ b/src/tests/lxc-test-unpriv
@@ -141,4 +141,22 @@ p1=$(run_cmd lxc-info -n c2 -p -H)
 
 run_cmd lxc-stop -n c2
 
+if which cgm >/dev/null 2>&1; then
+    echo "Testing containers under different cgroups per subsystem"
+    run_cmd cgm create freezer x1/x2
+    cgm movepid freezer x1 $$
+    run_cmd lxc-start -n c1 -d
+    p1=$(run_cmd lxc-info -n c1 -p -H)
+    [ "$p1" != "-1" ] || { echo "Failed to start container c1"; false; }
+    run_cmd lxc-info -n c1
+    run_cmd lxc-attach -n c1 -- /bin/true
+    run_cmd lxc-cgroup -n c1 freezer.state
+
+    echo "Testing lxc-attach and lxc-cgroup from different cgroup"
+    cgm movepid freezer x2 $$
+    run_cmd lxc-attach -n c1 -- /bin/true
+    run_cmd lxc-cgroup -n c1 freezer.state
+    run_cmd lxc-cgroup -n c1 memory.limit_in_bytes
+fi
+
 DONE=1
