From 94985d4e180160184a3493ed48ffce07d626160b Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 26 Mar 2020 15:32:29 +0100
Subject: dir: use cleanup macro in dir_mount()

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/storage/dir.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/lxc/storage/dir.c b/src/lxc/storage/dir.c
index d09e570ec..b657cc009 100644
--- a/src/lxc/storage/dir.c
+++ b/src/lxc/storage/dir.c
@@ -138,7 +138,7 @@ bool dir_detect(const char *path)
 
 int dir_mount(struct lxc_storage *bdev)
 {
-	char *mntdata = NULL;
+	__do_free char *mntdata = NULL;
 	unsigned long mflags = 0, mntflags = 0, pflags = 0;
 	int ret;
 	const char *src;
@@ -152,14 +152,12 @@ int dir_mount(struct lxc_storage *bdev)
 	ret = parse_mntopts(bdev->mntopts, &mntflags, &mntdata);
 	if (ret < 0) {
 		ERROR("Failed to parse mount options \"%s\"", bdev->mntopts);
-		free(mntdata);
 		return -EINVAL;
 	}
 
 	ret = parse_propagationopts(bdev->mntopts, &pflags);
 	if (ret < 0) {
 		ERROR("Failed to parse propagation options \"%s\"", bdev->mntopts);
-		free(mntdata);
 		return -EINVAL;
 	}
 
@@ -175,12 +173,10 @@ int dir_mount(struct lxc_storage *bdev)
 
 	if (ret < 0) {
 		SYSERROR("Failed to mount \"%s\" on \"%s\"", src, bdev->dest);
-		free(mntdata);
 		return -1;
 	}
 
 	TRACE("Mounted \"%s\" on \"%s\"", src, bdev->dest);
-	free(mntdata);
 	return ret;
 }
 
