From 8ff011a2aac8fd66a574dbdddc137d2a5c1a83fc Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sun, 19 Aug 2018 19:33:25 +0200
Subject: cmd: use utils.{c,h} helpers in lxc-usernsexec

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cmd/lxc_usernsexec.c | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/src/lxc/cmd/lxc_usernsexec.c b/src/lxc/cmd/lxc_usernsexec.c
index 35c00c8d..fa1b6a23 100644
--- a/src/lxc/cmd/lxc_usernsexec.c
+++ b/src/lxc/cmd/lxc_usernsexec.c
@@ -104,23 +104,13 @@ static int do_child(void *vargv)
 	char **argv = (char **)vargv;
 
 	/* Assume we want to become root */
-	ret = setgid(0);
-	if (ret < 0) {
-		CMD_SYSERROR("Failed to set gid to");
-		return -1;
-	}
-
-	ret = setuid(0);
-	if (ret < 0) {
-		CMD_SYSERROR("Failed to set uid to 0");
+	ret = lxc_switch_uid_gid(0, 0);
+	if (ret < 0)
 		return -1;
-	}
 
-	ret = setgroups(0, NULL);
-	if (ret < 0) {
-		CMD_SYSERROR("Failed to clear supplementary groups");
+	ret = lxc_setgroups(0, NULL);
+	if (ret < 0)
 		return -1;
-	}
 
 	ret = unshare(CLONE_NEWNS);
 	if (ret < 0) {
