Index: lxc-1.1.5/src/lxc/cgmanager.c
===================================================================
--- lxc-1.1.5.orig/src/lxc/cgmanager.c
+++ lxc-1.1.5/src/lxc/cgmanager.c
@@ -766,8 +766,8 @@ static char *try_get_abs_cgroup(const ch
 			NihError *nerr;
 			nerr = nih_error_get();
 			nih_free(nerr);
-		}
-		prune_init_scope(cgroup);
+		} else
+			prune_init_scope(cgroup);
 		return cgroup;
 	}
 
