From 6689cd63b2a5e54e48a4c46f71bebd37171d9bd3 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 22 Jan 2021 13:30:52 +0100
Subject: lxc_attach: include rexec conditionally

Cc: stable-4.0
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/tools/lxc_attach.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lxc/tools/lxc_attach.c b/src/lxc/tools/lxc_attach.c
index a8f493aa7..6d322ba0a 100644
--- a/src/lxc/tools/lxc_attach.c
+++ b/src/lxc/tools/lxc_attach.c
@@ -23,7 +23,9 @@
 #include "config.h"
 #include "confile.h"
 #include "log.h"
+#ifdef ENFORCE_MEMFD_REXEC
 #include "rexec.h"
+#endif
 #include "utils.h"
 
 lxc_log_define(lxc_attach, lxc);
