Index: lxc-1.1.5/src/lxc/cgmanager.c
===================================================================
--- lxc-1.1.5.orig/src/lxc/cgmanager.c
+++ lxc-1.1.5/src/lxc/cgmanager.c
@@ -183,11 +183,6 @@ static bool cgm_dbus_connect(void)
 	return true;
 }
 
-static bool cgm_supports_multiple_controllers;
-/*
- * if cgm_all_controllers_same is true, then cgm_supports_multiple_controllers
- * is true
- */
 static bool cgm_all_controllers_same;
 
 /*
@@ -207,11 +202,8 @@ static void check_supports_multiple_cont
 	size_t sz = 0;
 	char path[100];
 
-	cgm_supports_multiple_controllers = false;
 	cgm_all_controllers_same = false;
 
-	cgm_supports_multiple_controllers = true;
-
 	if (pid == -1)
 		sprintf(path, "/proc/self/cgroup");
 	else
@@ -433,7 +425,7 @@ static int chown_cgroup_wrapper(void *da
 	}
 	destuid = get_ns_uid(arg->origuid);
 
-	if (cgm_supports_multiple_controllers)
+	if (cgm_all_controllers_same)
 		slist = subsystems_inone;
 
 	for (i = 0; slist[i]; i++) {
@@ -491,7 +483,7 @@ static bool chown_cgroup(const char *cgr
 	 * This can't be done in the child namespace because it only group-owns
 	 * the cgroup
 	 */
-	if (cgm_supports_multiple_controllers)
+	if (cgm_all_controllers_same)
 		slist = subsystems_inone;
 
 	for (i = 0; slist[i]; i++) {
@@ -567,7 +559,7 @@ static void cgm_destroy(void *hdata)
 		return;
 	}
 
-	if (cgm_supports_multiple_controllers)
+	if (cgm_all_controllers_same)
 		slist = subsystems_inone;
 	for (i = 0; slist[i]; i++)
 		cgm_remove_cgroup(slist[i], d->cgroup_path);
@@ -587,7 +579,7 @@ static inline void cleanup_cgroups(char
 	int i;
 	char **slist = subsystems;
 
-	if (cgm_supports_multiple_controllers)
+	if (cgm_all_controllers_same)
 		slist = subsystems_inone;
 	for (i = 0; slist[i]; i++)
 		cgm_remove_cgroup(slist[i], path);
@@ -633,7 +625,7 @@ again:
 	}
 	existed = 0;
 
-	if (cgm_supports_multiple_controllers)
+	if (cgm_all_controllers_same)
 		slist = subsystems_inone;
 
 	for (i = 0; slist[i]; i++) {
@@ -1444,7 +1436,7 @@ struct cgroup_ops *cgm_ops_init(void)
 		return NULL;
 
 	if (api_version < CGM_SUPPORTS_MULT_CONTROLLERS)
-		cgm_supports_multiple_controllers = false;
+		cgm_all_controllers_same = false;
 
 	// if root, try to escape to root cgroup
 	if (geteuid() == 0 && !lxc_cgmanager_escape()) {
