From 0636c5b08a7c21576511d31c53b3e8b47dd3acce Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 9 Sep 2017 18:45:47 +0200
Subject: criu: use correct check initialization check

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/criu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/criu.c b/src/lxc/criu.c
index 676d759d..96688edc 100644
--- a/src/lxc/criu.c
+++ b/src/lxc/criu.c
@@ -551,7 +551,7 @@ static void exec_criu(struct criu_opts *opts)
 				external_not_veth = false;
 			}
 
-			if (n->name) {
+			if (n->name[0] != '\0') {
 				if (strlen(n->name) >= sizeof(eth))
 					goto err;
 				strncpy(eth, n->name, sizeof(eth));
