From 3a80a5a14e139dbcb21c6596a9cf880024d9cf1c Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 8 Apr 2020 14:42:05 +0200
Subject: seccomp: newer kernels require the buffer to be zeroed

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/seccomp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/seccomp.c b/src/lxc/seccomp.c
index 081d315ab..7820db8b2 100644
--- a/src/lxc/seccomp.c
+++ b/src/lxc/seccomp.c
@@ -1354,6 +1354,7 @@ int seccomp_notify_handler(int fd, uint32_t events, void *data,
 	char *cookie = conf->seccomp.notifier.cookie;
 	uint64_t req_id;
 
+	memset(req, 0, sizeof(*req));
 	ret = seccomp_notify_receive(fd, req);
 	if (ret) {
 		SYSERROR("Failed to read seccomp notification");
