From 834c7965c54e58db11c50bcc73edcbf3788f9e06 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 28 Jul 2021 18:39:31 +0200
Subject: conf: ensure devpts_fd is set to -EBADF

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/conf.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index d893328c7..482d1213b 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -3255,6 +3255,7 @@ struct lxc_conf *lxc_conf_init(void)
 	new->console.ptx = -EBADF;
 	new->console.pty = -EBADF;
 	new->console.name[0] = '\0';
+	new->devpts_fd = -EBADF;
 	memset(&new->console.ringbuf, 0, sizeof(struct lxc_ringbuf));
 	new->maincmd_fd = -1;
 	new->monitor_signal_pdeath = SIGKILL;
