From 9a2b673c425082980595c8745ce9c80d09d09f03 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sun, 19 Aug 2018 19:41:00 +0200
Subject: cmd: use safe number parsers in lxc-usernsexec

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cmd/lxc_usernsexec.c | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/src/lxc/cmd/lxc_usernsexec.c b/src/lxc/cmd/lxc_usernsexec.c
index a2c51800..4909a932 100644
--- a/src/lxc/cmd/lxc_usernsexec.c
+++ b/src/lxc/cmd/lxc_usernsexec.c
@@ -196,6 +196,7 @@ static int parse_map(char *map)
  */
 static int read_default_map(char *fnam, int which, char *user)
 {
+	int ret;
 	size_t len;
 	char *p1, *p2;
 	FILE *fin;
@@ -228,8 +229,20 @@ static int read_default_map(char *fnam, int which, char *user)
 			return -1;
 		}
 
-		newmap->hostid = atol(p1 + 1);
-		newmap->range = atol(p2 + 1);
+		ret = lxc_safe_ulong(p1 + 1, &newmap->hostid);
+		if (ret < 0) {
+			fclose(fin);
+			free(line);
+			return -1;
+		}
+
+		ret = lxc_safe_ulong(p2 + 1, &newmap->range);
+		if (ret < 0) {
+			fclose(fin);
+			free(line);
+			return -1;
+		}
+
 		newmap->nsid = 0;
 		newmap->idtype = which;
 
