From d6c739d3ea5febad82c6fe0eb8e07d0c6ebac2bc Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sun, 10 Sep 2017 06:42:10 +0200
Subject: utils: do not write to 0 sized buffer

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/utils.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/lxc/utils.c b/src/lxc/utils.c
index 07257d29..656d3ca8 100644
--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -2330,9 +2330,11 @@ int run_command(char *buf, size_t buf_size, int (*child_fn)(void *), void *args)
 	/* close the write-end of the pipe */
 	close(pipefd[1]);
 
-	bytes = read(pipefd[0], buf, (buf_size > 0) ? (buf_size - 1) : 0);
-	if (bytes > 0)
-		buf[bytes - 1] = '\0';
+	if (buf && buf_size > 0) {
+		bytes = read(pipefd[0], buf, buf_size - 1);
+		if (bytes > 0)
+			buf[bytes - 1] = '\0';
+	}
 
 	fret = wait_for_pid(child);
 	/* close the read-end of the pipe */
