From c65917f08c3391c0ebd7ecefa16a1252b3f38f07 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 27 Jun 2019 14:43:09 +0200
Subject: cgroups: move variable into tighter scope

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgfsng.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index ba0851601..473d108d9 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -1910,12 +1910,10 @@ __cgfsng_ops static int cgfsng_nrtasks(struct cgroup_ops *ops)
 __cgfsng_ops static bool cgfsng_escape(const struct cgroup_ops *ops,
 					 struct lxc_conf *conf)
 {
-	int i;
-
 	if (geteuid() || !ops->hierarchies)
 		return true;
 
-	for (i = 0; ops->hierarchies[i]; i++) {
+	for (int i = 0; ops->hierarchies[i]; i++) {
 		int ret;
 		__do_free char *fullpath = NULL;
 
