From 8750bfd3f20f880b9389d479cd3a6d38dc952f73 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 22 Jan 2021 19:16:06 +0100
Subject: configure: support static binaries

Including openssl handling.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 configure.ac | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index c3077c0e8..34f3ba31e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -278,9 +278,14 @@ if test "$enable_openssl" = "auto" ; then
 fi
 AM_CONDITIONAL([ENABLE_OPENSSL], [test "x$enable_openssl" = "xyes"])
 
+AC_ARG_ENABLE([static-binaries],
+	[AS_HELP_STRING([--enable-static-binaries], [build all binaries statically [default=no]])],
+	[enable_static_binaries=$enableval], [enable_static_binaries=no])
+AM_CONDITIONAL([ENABLE_STATIC_BINARIES], [test "x$enable_static_binaries" = "xyes"])
+
 AM_COND_IF([ENABLE_OPENSSL],
 	[AC_CHECK_HEADER([openssl/engine.h],[],[AC_MSG_ERROR([You must install the OpenSSL development package in order to compile lxc])])
-	AC_SUBST([OPENSSL_LIBS], '-lssl -lcrypto')])
+	AM_COND_IF([ENABLE_STATIC_BINARIES], [AC_SUBST([OPENSSL_LIBS], '-lssl -lcrypto -ldl')], [AC_SUBST([OPENSSL_LIBS], '-lssl -lcrypto')])])
 
 # SELinux
 AC_ARG_ENABLE([selinux],
@@ -433,6 +438,11 @@ AC_ARG_ENABLE([commands],
 	[enable_commands=$enableval], [enable_commands=yes])
 AM_CONDITIONAL([ENABLE_COMMANDS], [test "x$enable_commands" = "xyes"])
 
+AC_ARG_ENABLE([static-binaries],
+	[AS_HELP_STRING([--enable-static-binaries], [build all binaries statically [default=no]])],
+	[enable_static_binaries=$enableval], [enable_static_binaries=no])
+AM_CONDITIONAL([ENABLE_STATIC_BINARIES], [test "x$enable_static_binaries" = "xyes"])
+
 # Build with ASAN commands
 AC_ARG_ENABLE([asan],
 	[AS_HELP_STRING([--enable-asan], [build with address sanitizer enabled [default=no]])],
@@ -1040,6 +1050,8 @@ Binaries
    - lxc-user-nic:      $enable_commands
    - lxc-usernsexec:    $enable_commands
 
+ - static binaries: $enable_static_binaries
+
 Environment:
  - compiler: $CC
  - distribution: $with_distro
