From 939118118a0774d79b2371560ff1473e0c9dc7a0 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sun, 10 Sep 2017 08:01:31 +0200
Subject: lxc-unshare: do not pass NULL pointer

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/tools/lxc_unshare.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lxc/tools/lxc_unshare.c b/src/lxc/tools/lxc_unshare.c
index c294a608..25af9711 100644
--- a/src/lxc/tools/lxc_unshare.c
+++ b/src/lxc/tools/lxc_unshare.c
@@ -228,6 +228,9 @@ int main(int argc, char *argv[])
 	 *	dest: del + 1 == OUNT|PID
 	 *	src:  del + 3 == NT|PID
 	 */
+	if (!namespaces)
+		usage(argv[0]);
+
 	while ((del = strstr(namespaces, "MOUNT")))
 		memmove(del + 1, del + 3, strlen(del) - 2);
 
