From de85b03ef664caa04d66df8a01efa357e8e98635 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 29 Jul 2021 15:39:04 +0200
Subject: conf: set source property for devpts

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/conf.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 1b069e4a5..74da508d8 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -1679,6 +1679,10 @@ static int lxc_prepare_devpts_child(struct lxc_handler *handler)
 	if (fd_fs < 0)
 		return syserror("Failed to prepare filesystem context for devpts");
 
+	ret = fs_set_property(fd_fs, "source", "devpts");
+	if (ret < 0)
+		SYSTRACE("Failed to set \"source=devpts\" on devpts filesystem context %d", fd_fs);
+
 	ret = fs_set_property(fd_fs, "gid", "5");
 	if (ret < 0)
 		SYSTRACE("Failed to set \"gid=5\" on devpts filesystem context %d", fd_fs);
