From d767dcb2cffb9c8b518be686ebefd5568b188ec0 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 27 Jun 2019 14:48:34 +0200
Subject: cgroups: move variable into tighter scope

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgfsng.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index abdcccec3..a6e2d51b1 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -1544,7 +1544,7 @@ static int chowmod(char *path, uid_t chown_uid, gid_t chown_gid,
  */
 static int chown_cgroup_wrapper(void *data)
 {
-	int i, ret;
+	int ret;
 	uid_t destuid;
 	struct generic_userns_exec_data *arg = data;
 	uid_t nsuid = (arg->conf->root_nsuid_map != NULL) ? 0 : arg->conf->init_uid;
@@ -1574,7 +1574,7 @@ static int chown_cgroup_wrapper(void *data)
 	if (destuid == LXC_INVALID_UID)
 		destuid = 0;
 
-	for (i = 0; arg->hierarchies[i]; i++) {
+	for (int i = 0; arg->hierarchies[i]; i++) {
 		__do_free char *fullpath = NULL;
 		char *path = arg->hierarchies[i]->container_full_path;
 
