From 2d12b54f175fef1af879cda6ab2579bc4ceb8424 Mon Sep 17 00:00:00 2001
From: LiFeng <lifeng68@huawei.com>
Date: Thu, 27 Jun 2019 23:49:08 -0400
Subject: fix memory leak in do_storage_create

Signed-off-by: LiFeng <lifeng68@huawei.com>
---
 src/lxc/lxccontainer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/lxccontainer.c b/src/lxc/lxccontainer.c
index d6b88a5fd..3d864fd1c 100644
--- a/src/lxc/lxccontainer.c
+++ b/src/lxc/lxccontainer.c
@@ -1282,6 +1282,7 @@ static struct lxc_storage *do_storage_create(struct lxc_container *c,
 
 	if (!c->set_config_item(c, "lxc.rootfs.path", bdev->src)) {
 		ERROR("Failed to set \"lxc.rootfs.path = %s\"", bdev->src);
+		storage_put(bdev);
 		return NULL;
 	}
 
