From 5fc384856c413ce0c732f1c57235748884502580 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Fri, 27 Mar 2020 12:00:22 +0100
Subject: cgroups: remove unused variable

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgfsng.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index 4bcdf0c92..42cef8831 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -2118,7 +2118,6 @@ struct userns_exec_unified_attach_data {
 	const struct lxc_conf *conf;
 	int unified_fd;
 	pid_t pid;
-	uid_t origuid;
 };
 
 static int cgroup_unified_attach_wrapper(void *data)
