From 4db555c5876a118a642783d0400c05663d454fe0 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Mon, 8 Jul 2019 18:10:35 +0200
Subject: pidf_send_signal: fix return value

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 src/lxc/raw_syscalls.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lxc/raw_syscalls.c b/src/lxc/raw_syscalls.c
index f58b8d898..de8145e88 100644
--- a/src/lxc/raw_syscalls.c
+++ b/src/lxc/raw_syscalls.c
@@ -125,9 +125,9 @@ int lxc_raw_pidfd_send_signal(int pidfd, int sig, siginfo_t *info,
 			      unsigned int flags)
 {
 #ifdef __NR_pidfd_send_signal
-	syscall(__NR_pidfd_send_signal, pidfd, sig, info, flags);
+	return syscall(__NR_pidfd_send_signal, pidfd, sig, info, flags);
 #else
 	errno = ENOSYS;
-#endif
 	return -1;
+#endif
 }
