From fc72821b1be5863152cb26fabfb110d0028a515c Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 22 Aug 2018 13:27:24 +0200
Subject: log: fail build on ENFORCE_THREAD_SAFETY error

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/log.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lxc/log.h b/src/lxc/log.h
index b14ab1ce..a7f72b4c 100644
--- a/src/lxc/log.h
+++ b/src/lxc/log.h
@@ -353,6 +353,8 @@ ATTR_UNUSED static inline void LXC_##LEVEL(struct lxc_log_locinfo* locinfo,	\
 				(void)strerror_r(errno, errno_buf, sizeof(errno_buf));   \
 			}
 	#endif
+#elif ENFORCE_THREAD_SAFETY
+	#error ENFORCE_THREAD_SAFETY was set but cannot be guaranteed
 #else
 	#define lxc_log_strerror_r             \
 		char *ptr = NULL;              \
