From c45e33f09310d8015d2f61b6b883dfa1819004c8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@ubuntu.com>
Date: Mon, 7 Oct 2019 21:27:14 -0400
Subject: lxc.pc: Fix invalid @DLOG_LIBS@
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>
---
 lxc.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lxc.pc.in b/lxc.pc.in
index a60c70191..295ad663a 100644
--- a/lxc.pc.in
+++ b/lxc.pc.in
@@ -9,5 +9,5 @@ Description: linux container tools
 Version: @PACKAGE_VERSION@
 URL: http://linuxcontainers.org
 Libs: -L${libdir} -llxc -lutil
-Libs.private: @CAP_LIBS@ @SECCOMP_LIBS@ @OPENSSL_LIBS@ @SELINUX_LIBS@ @PAM_LIBS@ @DLOG_LIBS@
+Libs.private: @CAP_LIBS@ @SECCOMP_LIBS@ @OPENSSL_LIBS@ @SELINUX_LIBS@ @PAM_LIBS@
 Cflags: -I${includedir}
