From 3035eea6ce020660ecf621d7c24ea19ac8cd12d7 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Mon, 30 Mar 2020 18:16:16 +0200
Subject: start: log error when failing to create cgroup

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/start.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lxc/start.c b/src/lxc/start.c
index c8ebe7726..328516c97 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -1703,6 +1703,7 @@ static int lxc_spawn(struct lxc_handler *handler)
 	}
 
 	if (!cgroup_ops->payload_enter(cgroup_ops, handler)) {
+		ERROR("Failed to enter cgroups");
 		goto out_delete_net;
 	}
 
