From 3fa28c88c77fd5db119b1d7e5b47482914e6a1fb Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 27 Jan 2021 20:39:35 +0100
Subject: attach: coding style fixes

Cc: stable-4.0
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/attach.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index 3409a2091..7e5b4601b 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -1196,12 +1196,12 @@ int lxc_attach(struct lxc_container *container, lxc_attach_exec_t exec_function,
 		}
 
 		/* Create attached process. */
-		payload.ipc_socket = ipc_sockets[1];
-		payload.options = options;
-		payload.init_ctx = init_ctx;
+		payload.ipc_socket	= ipc_sockets[1];
+		payload.options		= options;
+		payload.init_ctx	= init_ctx;
 		payload.terminal_pts_fd = terminal.pty;
-		payload.exec_function = exec_function;
-		payload.exec_payload = exec_payload;
+		payload.exec_function	= exec_function;
+		payload.exec_payload	= exec_payload;
 
 		pid = lxc_raw_clone(CLONE_PARENT, NULL);
 		if (pid < 0) {
