From 98c3cc8022e6fe5ced656fac9f69ad17f969fa0f Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Tue, 31 Mar 2020 15:31:23 +0200
Subject: init: add ExecReload to lxc.service to only reload profiles

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
---
 config/init/systemd/lxc.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/config/init/systemd/lxc.service.in b/config/init/systemd/lxc.service.in
index 7d7bfb3ea..e3745dc2d 100644
--- a/config/init/systemd/lxc.service.in
+++ b/config/init/systemd/lxc.service.in
@@ -10,6 +10,7 @@ RemainAfterExit=yes
 ExecStartPre=@LIBEXECDIR@/lxc/lxc-apparmor-load
 ExecStart=@LIBEXECDIR@/lxc/lxc-containers start
 ExecStop=@LIBEXECDIR@/lxc/lxc-containers stop
+ExecReload=@LIBEXECDIR@/lxc/lxc-apparmor-load
 # Environment=BOOTUP=serial
 # Environment=CONSOLETYPE=serial
 Delegate=yes
