From b5a8b35c36f2ecb7cfb414f23615609612826c25 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 27 Jan 2021 20:42:21 +0100
Subject: attach: use free_disarm()

Cc: stable-4.0
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/attach.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index 7e5b4601b..861d2583c 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -109,8 +109,7 @@ static inline void lxc_proc_close_ns_fd(struct lxc_proc_context_info *ctx)
 
 static void lxc_proc_put_context_info(struct lxc_proc_context_info *ctx)
 {
-	free(ctx->lsm_label);
-	ctx->lsm_label = NULL;
+	free_disarm(ctx->lsm_label);
 
 	if (ctx->container) {
 		lxc_container_put(ctx->container);
