From ade86f1d3dffd0ead214309d933dc297d238b3f8 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 27 Jan 2021 20:52:08 +0100
Subject: attach: use close_prot_errno_disarm()

Cc: stable-4.0
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/attach.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index ebc3526c7..2be9fba02 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -813,8 +813,7 @@ __noreturn static void do_attach(struct attach_clone_payload *payload)
 			goto on_error;
 	}
 
-	close(payload->ipc_socket);
-	payload->ipc_socket = -EBADF;
+	close_prot_errno_disarm(payload->ipc_socket);
 	lxc_proc_put_context_info(init_ctx);
 	payload->init_ctx = NULL;
 
