From e5cd920f3af88518bf2e92f539ac904021c70930 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Wed, 27 Jan 2021 20:55:02 +0100
Subject: attach: add some DEBUG() logging to stdfd dpulication

Cc: stable-4.0
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/attach.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/lxc/attach.c b/src/lxc/attach.c
index 2be9fba02..bdf6da21f 100644
--- a/src/lxc/attach.c
+++ b/src/lxc/attach.c
@@ -827,13 +827,16 @@ __noreturn static void do_attach(struct attach_clone_payload *payload)
 	 * may want to make sure the fds are closed, for example.
 	 */
 	if (options->stdin_fd >= 0 && options->stdin_fd != STDIN_FILENO)
-		(void)dup2(options->stdin_fd, STDIN_FILENO);
+		if (dup2(options->stdin_fd, STDIN_FILENO))
+			DEBUG("Failed to replace stdin with %d", options->stdin_fd);
 
 	if (options->stdout_fd >= 0 && options->stdout_fd != STDOUT_FILENO)
-		(void)dup2(options->stdout_fd, STDOUT_FILENO);
+		if (dup2(options->stdout_fd, STDOUT_FILENO))
+			DEBUG("Failed to replace stdout with %d", options->stdin_fd);
 
 	if (options->stderr_fd >= 0 && options->stderr_fd != STDERR_FILENO)
-		(void)dup2(options->stderr_fd, STDERR_FILENO);
+		if (dup2(options->stderr_fd, STDERR_FILENO))
+			DEBUG("Failed to replace stderr with %d", options->stdin_fd);
 
 	/* close the old fds */
 	if (options->stdin_fd > STDERR_FILENO)
