From 27c7f7fde48976823e99cbc4231327f5de99ddd0 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Tue, 3 Aug 2021 09:22:46 +0200
Subject: conf: log failure to create tty mountpoint

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/conf.c | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 8dcfc60f3..37104761e 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -946,11 +946,20 @@ static int open_ttymnt_at(int dfd, const char *path)
 {
 	int fd;
 
-	fd = open_at(dfd, path, PROTECT_OPEN | O_CREAT | O_EXCL,
-		     PROTECT_LOOKUP_BENEATH, 0);
-	if (fd < 0 && (errno == ENXIO || errno == EEXIST))
-		fd = open_at(dfd, path, PROTECT_OPATH_FILE,
-			     PROTECT_LOOKUP_BENEATH, 0);
+	fd = open_at(dfd, path,
+		     PROTECT_OPEN | O_CREAT | O_EXCL,
+		     PROTECT_LOOKUP_BENEATH,
+		     0);
+	if (fd < 0) {
+		if (!IN_SET(errno, ENXIO, EEXIST))
+			return syserror("Failed to create \"%d/\%s\"", dfd, path);
+
+		SYSINFO("Failed to create \"%d/\%s\"", dfd, path);
+		fd = open_at(dfd, path,
+			     PROTECT_OPATH_FILE,
+			     PROTECT_LOOKUP_BENEATH,
+			     0);
+	}
 
 	return fd;
 }
