From bdb874b34c7c2196e83b71ad6f424e21d7cef970 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 4 Feb 2021 16:10:03 +0100
Subject: utils: fix mount_at()

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/utils.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/utils.c b/src/lxc/utils.c
index f960c1ee2..c48be5004 100644
--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -1253,7 +1253,7 @@ int mount_at(int dfd,
 		return -EIO;
 
 	if (!is_empty_string(src_buf))
-		ret = mount(src_under_dfd, dst_buf, fstype, mnt_flags, data);
+		ret = mount(src_buf, dst_buf, fstype, mnt_flags, data);
 	else
 		ret = mount(NULL, dst_buf, fstype, mnt_flags, data);
 
