From 06ff4ad40b3df436e442a88d81a471fc5223ed49 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Thu, 2 Apr 2020 18:19:31 +0200
Subject: lxccontainer: poll takes millisecond not seconds

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/lxccontainer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/lxccontainer.c b/src/lxc/lxccontainer.c
index 38439b80b..487d83894 100644
--- a/src/lxc/lxccontainer.c
+++ b/src/lxc/lxccontainer.c
@@ -2131,7 +2131,7 @@ static bool do_lxcapi_shutdown(struct lxc_container *c, int timeout)
 			 */
 
 			if (timeout != 0) {
-				ret = poll(&pidfd_poll, 1, timeout);
+				ret = poll(&pidfd_poll, 1, timeout * 1000);
 				if (ret < 0 || !(pidfd_poll.revents & POLLIN))
 					return false;
 
