From b371e5ab12bf211d10750db2d8e2ed3f4c6ac875 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 30 Jan 2021 00:55:19 +0100
Subject: croups: improve __do_bpf_program_free

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgroup2_devices.h | 17 +++--------------
 1 file changed, 3 insertions(+), 14 deletions(-)

diff --git a/src/lxc/cgroups/cgroup2_devices.h b/src/lxc/cgroups/cgroup2_devices.h
index 04f493e02..2da101ab4 100644
--- a/src/lxc/cgroups/cgroup2_devices.h
+++ b/src/lxc/cgroups/cgroup2_devices.h
@@ -19,6 +19,7 @@
 #include "compiler.h"
 #include "conf.h"
 #include "config.h"
+#include "memory_utils.h"
 #include "syscall_numbers.h"
 
 #ifdef HAVE_STRUCT_BPF_CGROUP_DEV_CTX
@@ -65,14 +66,6 @@ __hidden extern void bpf_program_free(struct bpf_program *prog);
 __hidden extern void bpf_device_program_free(struct cgroup_ops *ops);
 __hidden extern bool bpf_devices_cgroup_supported(void);
 
-static inline void __auto_bpf_program_free__(struct bpf_program **prog)
-{
-	if (*prog) {
-		bpf_program_free(*prog);
-		*prog = NULL;
-	}
-}
-
 __hidden extern int bpf_list_add_device(struct lxc_conf *conf, struct device_item *device);
 
 #else /* !HAVE_STRUCT_BPF_CGROUP_DEV_CTX */
@@ -129,10 +122,6 @@ static inline bool bpf_devices_cgroup_supported(void)
 	return false;
 }
 
-static inline void __auto_bpf_program_free__(struct bpf_program **prog)
-{
-}
-
 static inline int bpf_list_add_device(struct lxc_conf *conf,
 				      struct device_item *device)
 {
@@ -141,7 +130,7 @@ static inline int bpf_list_add_device(struct lxc_conf *conf,
 }
 #endif /* !HAVE_STRUCT_BPF_CGROUP_DEV_CTX */
 
-#define __do_bpf_program_free \
-	__attribute__((__cleanup__(__auto_bpf_program_free__)))
+define_cleanup_function(struct bpf_program *, bpf_program_free);
+#define __do_bpf_program_free call_cleaner(bpf_program_free)
 
 #endif /* __LXC_CGROUP2_DEVICES_H */
