From 25ced9d033fb2f68212c402b353e804b6f7ba0e3 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 30 Jan 2021 02:07:00 +0100
Subject: cgroups: don't initiliaze NULL log

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgroup2_devices.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/lxc/cgroups/cgroup2_devices.c b/src/lxc/cgroups/cgroup2_devices.c
index 26349bf8e..ef7d1ec9e 100644
--- a/src/lxc/cgroups/cgroup2_devices.c
+++ b/src/lxc/cgroups/cgroup2_devices.c
@@ -328,10 +328,8 @@ static int bpf_program_load_kernel(struct bpf_program *prog, char *log_buf,
 	if ((log_size != 0 && !log_buf) || (log_size == 0 && log_buf))
 		return ret_errno(EINVAL);
 
-	if (prog->kernel_fd >= 0) {
-		memset(log_buf, 0, log_size);
+	if (prog->kernel_fd >= 0)
 		return 0;
-	}
 
 	attr = &(union bpf_attr){
 		.prog_type	= prog->prog_type,
