From 0a155f47a871bdee0e57efddeccdaff8e55730a5 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 30 Jan 2021 11:02:10 +0100
Subject: log: remove pointless inline

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/log.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/log.c b/src/lxc/log.c
index 6c7574453..522a576ad 100644
--- a/src/lxc/log.c
+++ b/src/lxc/log.c
@@ -762,7 +762,7 @@ int lxc_log_set_level(int *dest, int level)
 	return 0;
 }
 
-inline int lxc_log_get_level(void)
+int lxc_log_get_level(void)
 {
 	return lxc_log_category_lxc.priority;
 }
