From 3388aa763a5d95be792b5f88074e86ed80cecd54 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 30 Jan 2021 11:09:45 +0100
Subject: seccomp: use lxc_log_get_fd()

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/seccomp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lxc/seccomp.c b/src/lxc/seccomp.c
index ba6811d90..9ea0ba9ab 100644
--- a/src/lxc/seccomp.c
+++ b/src/lxc/seccomp.c
@@ -1262,7 +1262,7 @@ int lxc_seccomp_load(struct lxc_conf *conf)
 #if HAVE_SCMP_FILTER_CTX
 	if ((lxc_log_get_level() <= LXC_LOG_LEVEL_TRACE ||
 	     conf->loglevel <= LXC_LOG_LEVEL_TRACE) &&
-	    lxc_log_fd >= 0) {
+	     lxc_log_get_fd() >= 0) {
 		ret = seccomp_export_pfc(conf->seccomp.seccomp_ctx, lxc_log_fd);
 		/* Just give an warning when export error */
 		if (ret < 0) {
