From 433a8578cbc19bbabf8bd69a1abe4e1d65e75ac9 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Sat, 30 Jan 2021 21:54:45 +0100
Subject: cgroups: align methods

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/cgroups/cgfsng.c | 54 ++++++++++++++++++++--------------------
 1 file changed, 27 insertions(+), 27 deletions(-)

diff --git a/src/lxc/cgroups/cgfsng.c b/src/lxc/cgroups/cgfsng.c
index 1eef175b8..fb46846fb 100644
--- a/src/lxc/cgroups/cgfsng.c
+++ b/src/lxc/cgroups/cgfsng.c
@@ -3454,33 +3454,33 @@ struct cgroup_ops *cgfsng_ops_init(struct lxc_conf *conf)
 	if (cg_init(cgfsng_ops, conf))
 		return NULL;
 
-	cgfsng_ops->data_init = cgfsng_data_init;
-	cgfsng_ops->payload_destroy = cgfsng_payload_destroy;
-	cgfsng_ops->monitor_destroy = cgfsng_monitor_destroy;
-	cgfsng_ops->monitor_create = cgfsng_monitor_create;
-	cgfsng_ops->monitor_enter = cgfsng_monitor_enter;
-	cgfsng_ops->monitor_delegate_controllers = cgfsng_monitor_delegate_controllers;
-	cgfsng_ops->payload_delegate_controllers = cgfsng_payload_delegate_controllers;
-	cgfsng_ops->payload_create = cgfsng_payload_create;
-	cgfsng_ops->payload_enter = cgfsng_payload_enter;
-	cgfsng_ops->payload_finalize = cgfsng_payload_finalize;
-	cgfsng_ops->escape = cgfsng_escape;
-	cgfsng_ops->num_hierarchies = cgfsng_num_hierarchies;
-	cgfsng_ops->get_hierarchies = cgfsng_get_hierarchies;
-	cgfsng_ops->get_cgroup = cgfsng_get_cgroup;
-	cgfsng_ops->get = cgfsng_get;
-	cgfsng_ops->set = cgfsng_set;
-	cgfsng_ops->freeze = cgfsng_freeze;
-	cgfsng_ops->unfreeze = cgfsng_unfreeze;
-	cgfsng_ops->setup_limits_legacy = cgfsng_setup_limits_legacy;
-	cgfsng_ops->setup_limits = cgfsng_setup_limits;
-	cgfsng_ops->driver = "cgfsng";
-	cgfsng_ops->version = "1.0.0";
-	cgfsng_ops->attach = cgfsng_attach;
-	cgfsng_ops->chown = cgfsng_chown;
-	cgfsng_ops->mount = cgfsng_mount;
-	cgfsng_ops->devices_activate = cgfsng_devices_activate;
-	cgfsng_ops->get_limiting_cgroup = cgfsng_get_limiting_cgroup;
+	cgfsng_ops->data_init				= cgfsng_data_init;
+	cgfsng_ops->payload_destroy			= cgfsng_payload_destroy;
+	cgfsng_ops->monitor_destroy			= cgfsng_monitor_destroy;
+	cgfsng_ops->monitor_create			= cgfsng_monitor_create;
+	cgfsng_ops->monitor_enter			= cgfsng_monitor_enter;
+	cgfsng_ops->monitor_delegate_controllers	= cgfsng_monitor_delegate_controllers;
+	cgfsng_ops->payload_delegate_controllers	= cgfsng_payload_delegate_controllers;
+	cgfsng_ops->payload_create			= cgfsng_payload_create;
+	cgfsng_ops->payload_enter			= cgfsng_payload_enter;
+	cgfsng_ops->payload_finalize			= cgfsng_payload_finalize;
+	cgfsng_ops->escape				= cgfsng_escape;
+	cgfsng_ops->num_hierarchies			= cgfsng_num_hierarchies;
+	cgfsng_ops->get_hierarchies			= cgfsng_get_hierarchies;
+	cgfsng_ops->get_cgroup				= cgfsng_get_cgroup;
+	cgfsng_ops->get					= cgfsng_get;
+	cgfsng_ops->set 				= cgfsng_set;
+	cgfsng_ops->freeze				= cgfsng_freeze;
+	cgfsng_ops->unfreeze				= cgfsng_unfreeze;
+	cgfsng_ops->setup_limits_legacy			= cgfsng_setup_limits_legacy;
+	cgfsng_ops->setup_limits			= cgfsng_setup_limits;
+	cgfsng_ops->driver				= "cgfsng";
+	cgfsng_ops->version				= "1.0.0";
+	cgfsng_ops->attach				= cgfsng_attach;
+	cgfsng_ops->chown				= cgfsng_chown;
+	cgfsng_ops->mount 				= cgfsng_mount;
+	cgfsng_ops->devices_activate			= cgfsng_devices_activate;
+	cgfsng_ops->get_limiting_cgroup			= cgfsng_get_limiting_cgroup;
 
 	return move_ptr(cgfsng_ops);
 }
