From 10c09b09e954eca74822d80e3904b697744239c9 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Mon, 1 Feb 2021 14:35:42 +0100
Subject: utils: use SYSTRACE() when logging stdio permission fixup failures

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/utils.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/lxc/utils.c b/src/lxc/utils.c
index c48be5004..fccd28d58 100644
--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -1961,15 +1961,15 @@ int fix_stdio_permissions(uid_t uid)
 
 		ret = fchown(std_fds[i], uid, st.st_gid);
 		if (ret) {
-			TRACE("Failed to chown standard I/O file descriptor %d to uid %d and gid %d",
-			      std_fds[i], uid, st.st_gid);
+			SYSTRACE("Failed to chown standard I/O file descriptor %d to uid %d and gid %d",
+			         std_fds[i], uid, st.st_gid);
 			fret = -1;
 			continue;
 		}
 
 		ret = fchmod(std_fds[i], 0700);
 		if (ret) {
-			TRACE("Failed to chmod standard I/O file descriptor %d", std_fds[i]);
+			SYSTRACE("Failed to chmod standard I/O file descriptor %d", std_fds[i]);
 			fret = -1;
 		}
 	}
