From fd20b43c4f728003419520aaca94c37590f80aaa Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@ubuntu.com>
Date: Mon, 1 Feb 2021 22:54:47 +0100
Subject: confile_utils: use lxc_log_trace()

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>
---
 src/lxc/confile_utils.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/lxc/confile_utils.c b/src/lxc/confile_utils.c
index 705d7e607..48b6970f6 100644
--- a/src/lxc/confile_utils.c
+++ b/src/lxc/confile_utils.c
@@ -225,8 +225,7 @@ void lxc_log_configured_netdevs(const struct lxc_conf *conf)
 	struct lxc_netdev *netdev;
 	struct lxc_list *it = (struct lxc_list *)&conf->network;;
 
-	if ((conf->loglevel != LXC_LOG_LEVEL_TRACE) &&
-	    (lxc_log_get_level() != LXC_LOG_LEVEL_TRACE))
+	if (!lxc_log_trace())
 		return;
 
 	if (lxc_list_empty(it)) {
